pkg/rule/rule_data.go

```diff
@@ -21,3 +23,16 @@ func NewRuleData(id []byte, exprs []expr.Any) RuleData {
 		ID:          id,
 	}
 }
+
+func (d RuleData) getCounters() (bytes *int64, packets *int64, error error) {
```

Comment:
Do we need to return pointers to these values or can we just return the values themeslves?

Also these won't be callable from the public API. Is it intended to be private or should anyone using the package be able to pull these values out of `RuleData`?

Also perhaps `Counters() (..., error)` is good enough and `Get*` is pretty much inferred?

Reply:
The decision to use a pointer here was to differentiate between having a counter be a valid 0 value and not finding a counter expression at all, so the values are `nil` rather than the default int64 value of 0 when not found. 

Anyone calling this could pull these values out of RuleData themselves since they have access to the expressions, but I'm not sure this is thorough enough to be an "officially supported implementation" for fetching counters for rules if that makes sense. 

I can be swayed on the naming to drop `get` lol

Reply:
Nice, thanks for that context! I didn't want to deny the PR based on any of my feedback so all is good with me then.

I generally shy away from any `Get*` calls in Go since it's very getter/setter-like in Java so I've leave it to you on keeping/dropping it, but wanted to call attention to it.

Reply:
I think that's totally fair
>it's very getter/setter-like in Java
Saw right through me haha I updated to drop `get`